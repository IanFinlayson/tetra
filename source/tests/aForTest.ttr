def main():
   vec = ["a","b","c","d","e"]
   print("Just made 'vec' and vec = ", vec, "\n")
   forTest(vec)
   print("Just returned from forTest() and vec = ", vec, "\n")
   if vec[0] == "a":
      print("vec[0] = 'a'","\n")
   elif vec[1] == "b":
      print("vec[1] = 'b'","\n")
   elif vec[2] == "qq" and vec[4] == "qq":
      print("vec[2] = 'qq' and vec[4] = 'qq'","\n")
   else:
      print("none of those things ^^", "\n")

def forTest(par [string]):
   print("Just entered forTest() and par = ", par, "\n")
   for i in [0 ... 4]:
      print("i = ", i, "\n")
      par[i] = "qq"
      print("Just set par[i] = 'qq' and par = ", par, "\n")
      if i == 2:
         print("in the i = 2 block and i = ", i, "\n")
         continue
      if i > 3:
         print("in the i > 3 block and i = ", i, "\n")
         break
      par[i] = "q"
      print("Just set par[i] = 'q' and par = ", par, "\n")
