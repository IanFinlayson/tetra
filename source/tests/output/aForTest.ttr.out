Just made 'vec' and vec = ['a', 'b', 'c', 'd', 'e']
Just entered forTest() and par = ['a', 'b', 'c', 'd', 'e']
i = 0
Just set par[i] = 'qq' and par = ['qq', 'b', 'c', 'd', 'e']
Just set par[i] = 'q' and par = ['q', 'b', 'c', 'd', 'e']
i = 1
Just set par[i] = 'qq' and par = ['q', 'qq', 'c', 'd', 'e']
Just set par[i] = 'q' and par = ['q', 'q', 'c', 'd', 'e']
i = 2
Just set par[i] = 'qq' and par = ['q', 'q', 'qq', 'd', 'e']
in the i = 2 block and i = 2
i = 3
Just set par[i] = 'qq' and par = ['q', 'q', 'qq', 'qq', 'e']
Just set par[i] = 'q' and par = ['q', 'q', 'qq', 'q', 'e']
i = 4
Just set par[i] = 'qq' and par = ['q', 'q', 'qq', 'q', 'qq']
in the i > 3 block and i = 4
Just returned from forTest() and vec = ['q', 'q', 'qq', 'q', 'qq']
vec[2] = 'qq' and vec[4] = 'qq'

